File: chap0/MallardDuck.java
public class MallardDuck extends Duck {
    public MallardDuck() {
        flyBehaviour = new FlyWithWings();
        quackBehaviour = new MuteQuack();
    };

    public void display() {
        System.out.println("I'm a real Mallard duck");
    }

}

File: chap0/MallardDuck.ts
import { Duck } from "./Duck";
import { FlyWithWings } from "./FlyWithWings";
import { MuteQuack } from "./MuteQuack";

export class MallardDuck extends Duck {
  constructor() {
    super();
    this.flyBehaviour = new FlyWithWings();
    this.quackBehaviour = new MuteQuack();
  }

  display(): void {
    console.log("I'm a real Mallard duck");
  }
}
